bin/all_sky_search/pycbc_template_kde_calc

```diff
@@ -53,40 +60,48 @@ def optimizedparam(sampleval, nfold=2):
-template_file = h5py.File(args.template_file, 'r')
-mass1_n = template_file['mass1'][:]
-mass2_n = template_file['mass2'][:]
-spin1z_n = template_file['spin1z'][:]
-spin2z_n = template_file['spin2z'][:]
-chi_eff_n = convert.chi_eff(mass1_n, mass2_n, spin1z_n, spin2z_n)
-mchirp_n = convert.mchirp_from_mass1_mass2(mass1_n, mass2_n)
-eta_n = convert.eta_from_mass1_mass2(mass1_n, mass2_n)
+#calling template data and constructing template_kde_hdf_file over same template params
+temp_file = h5py.File(args.template_file, 'r')
+mass1 = temp_file['mass1']
+tid = numpy.arange(len(mass1))  #tid : integer or array of int for Indices of the entries to be returned
+mass1, mass2, spin1z, spin2z = triggers.get_mass_spin(temp_file, tid)
+
+assert len(args.fit_param) == len(args.log_param)
+template_pars = []
+
+for param, slog in zip(args.fit_param, args.log_param):
+    data = triggers.get_param(param, args, mass1, mass2, spin1z, spin2z)
+    if slog in ['False']:
+        logging.info('Using param: %s', param)
+        template_pars.append(data)
+    elif slog in ['True']:
+        logging.info('Using log param: %s', param)
+        template_pars.append(numpy.log(data))
+    else:
+        raise ValueError("invalid log param argument, use 'True', or 'False'")
+sample1 = numpy.vstack((template_pars)).T
 
 if args.make_template_kde:
-    sample_n = numpy.vstack((numpy.log(mchirp_n), eta_n, chi_eff_n)).T
-    grid_pts_n = sample_n
     logging.info('Starting optimization of template KDE parameters')
-    optbw_n, optalpha_n = optimizedparam(sample_n, nfold=2)
+    optbw, optalpha = optimizedparam(sample1, nfold=eval(args.nfold_template))
     logging.info('Evaluating template KDE')
-    kde_data_n = kde_awkde(sample_n, grid_pts_n, alp=optalpha_n, gl_bandwidth=optbw_n)    
+    template_kde = kde_awkde(sample1, sample1, alp=optalpha, gl_bandwidth=optbw)
 
-    hdf = h5py.File(args.output_file, 'w')
-    hdf.create_dataset('kde_template', data=kde_data_n)
-    hdf.attrs['bank_kde'] = 'kde_template'
-    hdf.create_dataset('mass1', data=mass1_n)
-    hdf.create_dataset('mass2', data=mass2_n)
-    hdf.create_dataset('spin1z', data=spin1z_n)
-    hdf.create_dataset('spin2z', data=spin2z_n)
-    hdf.create_dataset('mchirp', data=mchirp_n)
-    hdf.create_dataset('eta', data=eta_n)
-    hdf.create_dataset('chi_eff', data=chi_eff_n)
-    hdf.create_dataset('grid_pts', data=grid_pts_n)
-    hdf.close()
+    with h5py.File(args.output_file, "w") as f_dest:
+        g1 = f_dest.create_group("new_data")
+        g1.create_dataset("template_kde", data=template_kde)
+        g1.create_dataset("grid", data=sample1)
+        for name in args.fit_param:
+            g1.create_dataset(name, data= triggers.get_param(name, args, mass1, mass2, spin1z, spin2z))
+        g1.attrs['fit_param'] = args.fit_param
+        g1.attrs['log_param'] = args.log_param
+        with h5py.File("sbbh2.hdf", "r") as f_src:
```

Comment:
I am not seeing why you have put the file name in the code here. All the files to be used by the code should be given as input options. This code should be able to be run with _any_ bank file, not just one called 'sbbh2.hdf'.